### PR TITLE
UCS/TOPO: Add methods for number of devices and distance string

### PR DESCRIPTION
**Pull request by yosefe**
# Why
- Return total number of system devices, for allocating temporary distance arrays
- Convert distance to human string for debug purpose
- Common definition of "default" distance: zero latency, infinite bandwidth

### CODE REVIEW

**Comment by dmitrygx** [comment]
bot:pipe:retest

File: src/ucs/sys/topo.c [hunk 1]
```diff
@@ -38,6 +38,10 @@ typedef struct ucs_topo_global_ctx {
 } ucs_topo_global_ctx_t;
 
 
+ucs_sys_dev_distance_t ucs_topo_default_distance = {
```
Comment (dmitrygx) [inline src/ucs/sys/topo.c:41]: `const`?

File: src/ucs/sys/topo.h [hunk 2]
```diff
@@ -99,6 +118,14 @@ ucs_status_t
 ucs_topo_find_device_by_bdf_name(const char *name, ucs_sys_device_t *sys_dev);
 
 
+/**
+ * Get the number of registered system devices
```
Comment (dmitrygx) [inline src/ucs/sys/topo.h:122]: add a trailing point

File: src/ucs/sys/topo.h [hunk 3]
```diff
@@ -99,6 +118,14 @@ ucs_status_t
...
+ *
+ * @return Number of system devices
```
Comment (dmitrygx) [inline src/ucs/sys/topo.h:124]: add a trailing point

File: src/ucs/sys/topo.h [hunk 4]
```diff
@@ -75,6 +81,19 @@ ucs_status_t ucs_topo_get_distance(ucs_sys_device_t device1,
...
 
 
+/**
+ * Convert the distance to a human-readable string.
+ *
+ * @param [in]  distance   Distance between two devices
+ * @param [out] buffer     String buffer to fill with distance string
+ * @param [in]  max        Maximal size of the string buffer.
+ *
+ * @return Pointer to the distance string
```
Comment (dmitrygx) [inline src/ucs/sys/topo.h:91]: add a trailing point

**Review by dmitrygx — COMMENTED** [review]
👍 besides minor comments

**Review by hoopoepg — APPROVED** [review]

**Review by dmitrygx — APPROVED** [review]

**Pull request merged**